baseplate/integration/wrapped_context.py

```diff
@@ -0,0 +1,21 @@
+import copy
+
+
+def merge_two_dicts(x, y):
+    """Given two dicts, merge them into a new dict as a shallow copy."""
+    z = x.copy()
+    z.update(y)
+    return z
+
+
+class WrappedRequestContext(object):
+    def __init__(self, context, trace=None):
+        self._context = context
+        self.__dict__.update(self._context.__dict__)
+
+    def __getattr__(self, attr):
+        return getattr(self._context, attr)
+
+    def clone(self):
+        new_context = copy.copy(self._context)
```

Comment:
do we actually want to copy the underlying object ever? or just keep referencing it since we're just a neat wrapper.

i'm concerned that we'd do something like

```python
with request.trace.make_local("...") as (local_request, local):
    local_request.response_code = 302
```

and that would get lost because we were copying the underlying request object instead of just referencing.

(it'd probably never happen so dumbly in real code, but i could see injecting a local span around a block of code that passes a request object around various function calls getting messed up)